### ISSUE TITLE
Connection error with IWA/SSPI

### ISSUE THREAD

**Issue by jgoodleaf** [sassoftware/saspy#141]
saspy version 2.24

When trying to create a SASsession instance, saspy appears not to pick up any authentication information. In the error below, you can see `user ID "" or the password is incorrect.`

So first, sspiauth.dll is definitely in `PATH`. (And there's no error about checking that.) 

I have two operative configs in sascfg_personal, which I call `winiomwin` and `winiomIWA`, following convention. If I make the session with `winiomwin`, I am prompted for a username and password, and when those are given, it connects successfully. If I use the IWA, I get the following error. The only difference between those configs is that the IWA has `sspi: True`. 
```
The application could not log on to the server "<SASHOST>:8591". The user ID "" or the password is incorrect.

SAS process has terminated unexpectedly. RC from wait was: 4294967290
SAS Connection failed. No connection established. Double check you settings in sascfg.py file.

Attempted to run program java with the following parameters:['java', '-classpath', 'C:\\SASJAR\\products\\deploywiz__94330__prt__xx__sp0__1\\deploywiz\\sas.svc.connection.jar;C:\\SASJAR\\products\\deploywiz__94330__prt__xx__sp0__1\\deploywiz\\log4j.jar;C:\\SASJAR\\products\\deploywiz__94330__prt__xx__sp0__1\\deploywiz\\sas.security.sspi.jar;C:\\SASJAR\\products\\deploywiz__94330__prt__xx__sp0__1\\deploywiz\\sas.core.jar;C:\\Anaconda\\envs\\sas\\Lib\\site-packages\\saspy\\java\\saspyiom.jar', 'pyiom.saspy2j', '-host', 'localhost', '-stdinport', '49707', '-stdoutport', '49708', '-stderrport', '49709', '-iomhost', '<host>', '-iomport', '8591', '-spn', '']

No SAS process attached. SAS process has terminated unexpectedly.
No SAS process attached. SAS process has terminated unexpectedly.
No SAS process attached. SAS process has terminated unexpectedly.
```
Integrated authentication appears otherwise to be working, as all my other Windows foofaraw connects up just fine with Active Directory and whatnot.

**Comment by tomweber-sas** [comment]
Hmmm, Can you show me the two config definitions (obfuscating any specifics in there is fine)?
The workspace server your connecting to (or maybe object spawner), has to be configured to use IWA, though you mention that you are connecting to that server via IWA with other SAS clients?
And when providing user/pw, it does work, which is good, and means you're configured right. 

I'll have to check w/ some folks here to see how this would happen given the situation your describing. Is the workspace server Windows or Linux? 

Thanks,
Tom

**Comment by jgoodleaf** [comment]
```
winiomwin  = {'java'    : 'java',
            'iomhost'   : 'hostname',
            'iomport'   : 8591,
            'encoding'  : 'windows-1252',
            'classpath' : cpW,
            #'sspi': True
            }

winiomIWA  = {'java'    : 'java',
            'iomhost'   : 'hostname',
            'iomport'   : 8591,
            'encoding'  : 'windows-1252',
            'classpath' : cpW,
            'sspi': True
            }
```
And the `cpW` variable is:
```
cpW  =  "C:\SASJAR\products\deploywiz__94330__prt__xx__sp0__1\deploywiz\sas.svc.connection.jar"
cpW += ";C:\SASJAR\products\deploywiz__94330__prt__xx__sp0__1\deploywiz\log4j.jar"
cpW += ";C:\SASJAR\products\deploywiz__94330__prt__xx__sp0__1\deploywiz\sas.security.sspi.jar"
cpW += ";C:\SASJAR\products\deploywiz__94330__prt__xx__sp0__1\deploywiz\sas.core.jar"
cpW += ";C:\Anaconda\envs\saspy2\Lib\site-packages\saspy\java\saspyiom.jar"
```
I'll double-check, but I am fairly sure our SAS programmers are using IWA to connect (I come at this from the Python side; I am not a SAS user). The IOM host is on Windows.

**Comment by jgoodleaf** [comment]
WAIT! I figured it out I think. When we did the setup, we copied the classpath value and it is pointing at a different version of the `saspyiom.jar` file--from an earlier virtualenv with an older version of saspy. When I change the classpath to point to the `saspyiom.jar` file from a fresh installation of `saspy` (I created a new virtualenv) it works!

So the moral of the story here is: **Don't copy-paste the `classpath` value just to save time!**

Thanks for your help. I think I only spotted it because you asked me to copy in the configs.
